src/functions/leastSquares.jl

```diff
@@ -27,27 +27,44 @@ type LeastSquares{R <: RealOrComplex} <: ProximableConvex
       new(A, b, A'*b, lambda, -1, A*A')
     end
   end
+
+  function LeastSquares{RC,R,M,V}(A::M, b::V, lambda::R) where {RC <: RealOrComplex, R<:Real, I<:Integer, M<:SparseMatrixCSC{RC,I}, V<:AbstractArray{RC,1}}
+    LeastSquares{RC,R,M,V,SparseArrays.CHOLMOD.Factor{RC}}(A,b,lambda)
+  end
+  function LeastSquares{RC,R,M,V}(A::M, b::V, lambda::R) where {RC <: RealOrComplex, R<:Real, M<:DenseArray{RC,2}, V<:AbstractArray{RC,1}}
+    LeastSquares{RC,R,M,V,LinAlg.Cholesky{RC,Array{RC,2}}}(A,b,lambda)
+  end
+  function LeastSquares{RC,R,M,V}(A::M, b::V, lambda::R) where {RC <: AbstractArray, R<:Real, M<:AbstractArray{RC,2}, V<:AbstractArray{RC,1}}
+    warn("Could not infer type of Factorization for $M in LeastSquares, this type will be type-unstable")
+    LeastSquares{RC,R,M,V,Factorization}(A,b,lambda)
+  end
```

Comment:
This is definitely quite ugly, but does improve performance by having type stable constructors and making sure the types are not abstract.